core/server/namespace_test.go

```diff
@@ -59,3 +59,40 @@ func TestGetFluxNamespace_notFound(t *testing.T) {
 	_, err = coreClient.GetFluxNamespace(ctx, &pb.GetFluxNamespaceRequest{})
 	g.Expect(err).To(HaveOccurred())
 }
+
+func TestListNamespaces(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	ctx := context.Background()
+
+	coreClient, cleanup := makeGRPCServer(k8sEnv.Rest, t)
```

Comment:
You can simplify this by adding the cleanup with [t.Cleanup(func(){...})](https://pkg.go.dev/testing#T.Cleanup) in `makeGRPCServer`?

Reply:
I don't think it will work as we want. Last time I had very hard time with namespaces in tests. A `Delete` call did not delete the resources, or did something different because the next test case could query the "deleted" namespace. At the end, there I used `Update` to remove a label from a namespace. It can be useful to improve, but not here.

Reply:
Can you point to this code?

This doesn't sound quite right, it can take a while to delete a namespace.

But, I don't see how this is relevant to the issue of simplifying the API?

Reply:
https://github.com/weaveworks/weave-gitops/pull/1515/files#diff-f66d7854f580d2c66e5671211c392900713f66b468390f99207eb5395bb0daf3R35-R40

Reply:
Oh and now I see, I though you commented on a different line.

Reply:
https://github.com/weaveworks/weave-gitops/issues/1649